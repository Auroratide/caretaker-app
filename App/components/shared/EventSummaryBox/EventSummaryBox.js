import React from 'react';
import { Text, View } from 'react-native';
import styles from './style';

export default function EventSummaryBox( { name, volunteer } ) {
  return <View style={styles.container}>
    <Text style={[styles.text, styles.name]}>{name}</Text>
    <Text style={[styles.text, styles.volunteer]}>{volunteer}</Text>
  </View>;
}
