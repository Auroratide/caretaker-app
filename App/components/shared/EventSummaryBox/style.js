import { StyleSheet } from 'react-native';
import theme from '../../../config/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: theme.palette.secondaryColor,
    borderRadius: 2,
    padding: 4
  },
  text: {
    color: theme.palette.secondaryColor
  },
  name: {
    fontWeight: 'bold'
  },
  volunteer: {}
});

export default styles;
