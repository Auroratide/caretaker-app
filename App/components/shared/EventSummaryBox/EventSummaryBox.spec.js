import React from 'react';
import renderer from 'react-test-renderer';
import EventSummaryBox from './EventSummaryBox';

describe('<EventSummaryBox />', () => {
  let props;
  const eventSummaryBox = () => renderer.create(<EventSummaryBox {...props} />);

  beforeEach(() => {
    props = {
      name: undefined,
      volunteer: undefined
    };
  });

  it('should render correctly', () => {
    props.name = 'Chemotherapy';
    props.volunteer = 'Adept Wilkenson';
    expect(eventSummaryBox()).toMatchSnapshot();
  });

});
