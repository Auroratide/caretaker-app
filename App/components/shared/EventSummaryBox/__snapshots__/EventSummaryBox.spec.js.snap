// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventSummaryBox /> should render correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#E66969",
      "borderRadius": 2,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flex": 1,
      "padding": 4,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#E66969",
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Chemotherapy
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#E66969",
        },
        Object {},
      ]
    }
  >
    Adept Wilkenson
  </Text>
</View>
`;
